packages/code-utils/src/index.ts

```diff
@@ -76,17 +95,15 @@ export function parseCode(
   return instructions;
 }
 
-export default {
-  //for compatibility
-  parseCode
-};
-
 function isValidCBOR(metadata: Uint8Array) {
   try {
     //attempt to decode but discard the value
     //note this *will* throw if there's data left over,
     //which is what we want it to do
-    cbor.decodeFirstSync(metadata);
+    //HACK: this version of cbor doesn't accept Uint8Arrays,
+    //but it does accept Buffers.  (Unfortunately newer versions
+    //cause problems. :-/ )
+    cbor.decodeFirstSync(Buffer.from(metadata));
```

Comment:
What problems does the latest `cbor` cause? My gut feel on things like this is that it's usually best to go for the latest version of a package unless it's just broken, as we're likely to want to move to that version later on for various reasons anyway (e.g. due to CVEs, etc).

That said, if we revert the minor version change in this PR I suppose that's fine, too.

Reply:
Alternatively is it possible to refactor the code above to use `Buffer` instead of `UInt8Array` so that we don't need the hack here? If you think doing so would be painful, cause performance or readability issues, please disregard.

Reply:
Also just to be clear I'm not suggesting that you must either revert back to 5.1.0 or use 7.x. I think saying "if you change the version you _must_ do so in the most optimal fashion possible" is a bit pennywise, pound foolish.

Just mentioned all of the above in case you encountered the issues with 7.x and "noped out" of the change before determining whether it'd be high or low effort to fix said issues.

Reply:
I don't actually remember offhand what the problems with 7.x were, good question.  I think they were build problems in `@truffle/contract`, similar but not identical to what I got with 8.x?  I'd have to try it again to be certain, though.

Reply:
Ah sorry, I keep saying `7.x` when I mean whatever the latest version is. My apologies if that's `8x` or something else.

Reply:
OK, here's the error message:
```
@truffle/contract: $ yarn compile
@truffle/contract: $ NODE_ENV=production webpack --config ./webpack.config.js && NODE_ENV=development webpack --config ./webpack.config.js
@truffle/contract: assets by status 7.81 MiB [cached] 1 asset
@truffle/contract: orphan modules 2.42 MiB [orphan] 483 modules
@truffle/contract: runtime modules 1.28 KiB 7 modules
@truffle/contract: modules by path ../../node_modules/ 11.9 MiB 2821 modules
@truffle/contract: modules by path ../codec/ 928 KiB 297 modules
@truffle/contract: modules by path ../contract-schema/ 234 KiB 47 modules
@truffle/contract: modules by path ./ 416 KiB 23 modules
@truffle/contract: modules by path ../interface-adapter/ 285 KiB 22 modules
@truffle/contract: optional modules 300 bytes [optional] 20 modules
@truffle/contract: modules by path ../debug-utils/ 152 KiB 10 modules
@truffle/contract: modules by path ../compile-common/dist/src/ 11.8 KiB 8 modules
@truffle/contract: modules by path ../abi-utils/dist/lib/*.js 13.9 KiB
@truffle/contract:   ../abi-utils/dist/lib/index.js 1.34 KiB [built] [code generated]
@truffle/contract:   + 3 modules
@truffle/contract: + 7 modules
@truffle/contract: ERROR in ../../node_modules/cbor/lib/utils.js 22:9-42
@truffle/contract: Module not found: Error: Can't resolve 'node-inspect-extracted' in '/home/sniffnoy/truffle/truffle/node_modules/cbor/lib'
@truffle/contract: resolve 'node-inspect-extracted' in '/home/sniffnoy/truffle/truffle/node_modules/cbor/lib'
@truffle/contract:   Parsed request is a module
@truffle/contract:   using description file: /home/sniffnoy/truffle/truffle/node_modules/cbor/package.json (relative path: ./lib)
@truffle/contract:     resolve as module
@truffle/contract:       /home/sniffnoy/truffle/truffle/node_modules/cbor/lib/node_modules doesn't exist or is not a directory
@truffle/contract:       /home/sniffnoy/truffle/truffle/node_modules/cbor/node_modules doesn't exist or is not a directory
@truffle/contract:       /home/sniffnoy/truffle/truffle/node_modules/node_modules doesn't exist or is not a directory
@truffle/contract:       looking for modules in /home/sniffnoy/truffle/truffle/node_modules
@truffle/contract:         single file module
@truffle/contract:           using description file: /home/sniffnoy/truffle/truffle/package.json (relative path: ./node_modules/node-inspect-extracted)
@truffle/contract:             no extension
@truffle/contract:               Field 'browser' doesn't contain a valid alias configuration
@truffle/contract:               /home/sniffnoy/truffle/truffle/node_modules/node-inspect-extracted doesn't exist
@truffle/contract:             .js
@truffle/contract:               Field 'browser' doesn't contain a valid alias configuration
@truffle/contract:               /home/sniffnoy/truffle/truffle/node_modules/node-inspect-extracted.js doesn't exist
@truffle/contract:             .json
@truffle/contract:               Field 'browser' doesn't contain a valid alias configuration
@truffle/contract:               /home/sniffnoy/truffle/truffle/node_modules/node-inspect-extracted.json doesn't exist
@truffle/contract:             .wasm
@truffle/contract:               Field 'browser' doesn't contain a valid alias configuration
@truffle/contract:               /home/sniffnoy/truffle/truffle/node_modules/node-inspect-extracted.wasm doesn't exist
@truffle/contract:         /home/sniffnoy/truffle/truffle/node_modules/node-inspect-extracted doesn't exist
@truffle/contract:       /home/sniffnoy/truffle/node_modules doesn't exist or is not a directory
@truffle/contract:       looking for modules in /home/sniffnoy/node_modules
@truffle/contract:         single file module
@truffle/contract:           No description file found in /home/sniffnoy/node_modules or above
@truffle/contract:           no extension
@truffle/contract:             /home/sniffnoy/node_modules/node-inspect-extracted doesn't exist
@truffle/contract:           .js
@truffle/contract:             /home/sniffnoy/node_modules/node-inspect-extracted.js doesn't exist
@truffle/contract:           .json
@truffle/contract:             /home/sniffnoy/node_modules/node-inspect-extracted.json doesn't exist
@truffle/contract:           .wasm
@truffle/contract:             /home/sniffnoy/node_modules/node-inspect-extracted.wasm doesn't exist
@truffle/contract:         /home/sniffnoy/node_modules/node-inspect-extracted doesn't exist
@truffle/contract:       /home/node_modules doesn't exist or is not a directory
@truffle/contract:       /node_modules doesn't exist or is not a directory
@truffle/contract:  @ ../../node_modules/cbor/lib/decoder.js 6:14-32
@truffle/contract:  @ ../../node_modules/cbor/lib/cbor.js 6:0-38
@truffle/contract:  @ ../codec/dist/lib/contexts/utils.js 31:26-41
@truffle/contract:  @ ../codec/dist/lib/contexts/index.js 29:27-45
@truffle/contract:  @ ../codec/dist/lib/index.js 139:30-51
@truffle/contract:  @ ../debug-utils/index.js 4:14-39
@truffle/contract:  @ ./lib/promievent.js 2:19-50
@truffle/contract:  @ ./lib/execute.js 2:19-42
@truffle/contract:  @ ./lib/contract/index.js 4:16-37
@truffle/contract:  @ ./index.js 2:17-42
@truffle/contract: webpack 5.73.0 compiled with 1 error in 90166 ms
```

Reply:
> Also if we have to stay on 5.1.0 or 5.2.0, is it possible to refactor the code above to use Buffer instead of UInt8Array so that we don't need the hack here? If you think doing so would be painful, cause performance or readability issues, please disregard.

No, it would be a pretty straightforward change.  I could totally go do that.

Reply:
I guess there's some extent to which I'm wary of expanding the use of the Node-specific Buffer class... but I guess I'm using it anyway here, and we use it in other places without problem, so I guess it doesn't really matter. :P

Reply:
This would be my suggestion if you even want to make the change at all.

I don't think eliminating the `UInt8Array` adds any functional value, and the minor performance and code quality improvements that we get for the change are arguably quite low value by comparison to the rest of the changes here, so I see no reason to block the merge of this PR.

Reply:
Re: the webpack error above, I think solving that is also not worth doing in this PR.

That said, without having looked at the `cbor` lib at all, I'd bet an arbitrary modest sum that the error in question is webpack choking on a conditional require for a missing (hopefully optional) peer dependency, and you very likely can add `node-expect-extracted` to the list of externals in the webpack config to resolve it.

You'll need to have a look at the code to verify that's the case, however.

Reply:
Interesting.  Yeah I'm not very familiar with webpack errors.

Yeah 8.x gave a different build error which I didn't post.  But I can post that error message too if people care...

Reply:
FWIW, here's the error we get on 8.x:

```
@truffle/code-utils: $ yarn build
@truffle/code-utils: $ tsc
@truffle/code-utils: ../../node_modules/cbor/types/lib/encoder.d.ts(102,40): error TS2304: Cannot find name 'URL'.
@truffle/code-utils: error Command failed with exit code 2.
@truffle/code-utils: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@truffle/code-utils: error Command failed with exit code 2.
@truffle/code-utils: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Oh interesting -- I'm trying different 7.x versions and they don't all lead to that error!  Like 7.0.0 doesn't.  It does bloat the bundle though. :-/  Oh well.